### PR TITLE
#27124: Update wishlist image logic to match logic on wishlist page

### PR DESCRIPTION
**Pull request by mtbottens**
### Description (*)

Updates image logic on the "Share wishlist" email to match the image logic in the my account wishlist page.

### Related Pull Requests

N/A

### Fixed Issues (if relevant)

1. magento/magento2#27124: Share Wishlist Email: Image Logic Issue

### Manual testing scenarios (*)

1. Set Stores > Configuration > Sales > Checkout > Shopping Cart > Configurable Product Image = "Product Thumbnail Itself"
2. Add "Olivia 1/4 Zip Light Jacket" (Size = Medium, Color = Black) to wishlist
3. Navigate to wishlist page, and click the share wishlist button
4. Fill in the form
5. Share the wishlist
6. Observe that the wishlist email is using the configurable product image

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Review by eduard13 — CHANGES_REQUESTED** [review]
Hi @mtbottens, thank you for your contribution. However, for me it looks like the problem isn't related only with the image, but with the product URL as well.
So, may I ask you to update also the URL that leads to product page to auto select the configurable attributes (exactly how we have it on Wishlist page)?

Thank you.

**Comment by mtbottens** [comment]
Hey @eduard13, I've updated this pull request to handle the links as mentioned.

Thanks!

File: app/code/Magento/Wishlist/Block/Share/Email/Items.php [hunk 1]
```diff
@@ -11,17 +11,57 @@
...
+     * @param \Magento\Framework\App\Http\Context $httpContext
+     * @param ItemResolverInterface $itemResolver
+     * @param array $data
+     * @param ConfigInterface|null $config
+     * @param UrlBuilder|null $urlBuilder
+     */
+    public function __construct(
+        \Magento\Catalog\Block\Product\Context $context,
+        \Magento\Framework\App\Http\Context $httpContext,
+        ItemResolverInterface $itemResolver,
```
Comment (eduard13) [inline app/code/Magento/Wishlist/Block/Share/Email/Items.php]: Could you please inject the new constructor dependency with a fallback to the ObjectManger?
https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/#adding-a-constructor-parameter

File: app/code/Magento/Wishlist/Block/Share/Email/Items.php [hunk 2]
```diff
@@ -11,17 +11,57 @@
...
+use Magento\Framework\View\ConfigInterface;
+use Magento\Wishlist\Model\Item;
+
 /**
  * @api
  * @since 100.0.2
  */
 class Items extends \Magento\Wishlist\Block\AbstractBlock
 {
+    /** @var ItemResolverInterface */
```
Comment (eduard13) [inline app/code/Magento/Wishlist/Block/Share/Email/Items.php]: Please move the docblock to a new line.
```suggestion
    /**
     * @var ItemResolverInterface
     */
```

File: app/code/Magento/Wishlist/Block/Share/Email/Items.php [hunk 3]
```diff
@@ -11,17 +11,57 @@
...
+    public function __construct(
+        \Magento\Catalog\Block\Product\Context $context,
+        \Magento\Framework\App\Http\Context $httpContext,
+        ItemResolverInterface $itemResolver,
+        array $data = [],
+        ConfigInterface $config = null,
+        UrlBuilder $urlBuilder = null
+    ) {
+        $this->itemResolver = $itemResolver;
+        parent::__construct($context, $httpContext, $data, $config, $urlBuilder);
```
Comment (eduard13) [inline app/code/Magento/Wishlist/Block/Share/Email/Items.php]: Please make sure that you call the parent construct first.
```suggestion
        parent::__construct($context, $httpContext, $data, $config, $urlBuilder);
        $this->itemResolver = $itemResolver;
```

File: app/code/Magento/Wishlist/Block/Share/Email/Items.php [hunk 4]
```diff
@@ -11,17 +11,57 @@
...
+    }
+
+    /**
+     * Identify the product from which thumbnail should be taken.
+     *
+     * @param Item $item
+     * @return Product
+     */
+    public function getProductForThumbnail(Item $item) : Product
```
Comment (eduard13) [inline app/code/Magento/Wishlist/Block/Share/Email/Items.php]:
```suggestion
    public function getProductForThumbnail(Item $item): Product
```

**Review by eduard13 — CHANGES_REQUESTED** [review]
Thank you for your updates, could you please check the review notes and update the code based on them?
Also, could you please fix the failing static tests as well?
Thank you.

**Comment by mtbottens** [comment]
@eduard13 Thank you for the code review, I have made the requested changes.

File: app/code/Magento/Wishlist/Block/Share/Email/Items.php [hunk 5]
```diff
@@ -11,17 +11,61 @@
...
+    ) {
+        parent::__construct($context, $httpContext, $data, $config, $urlBuilder);
+        $this->itemResolver = $itemResolver ?? ObjectManager::getInstance()->get(ItemResolverInterface::class);
+    }
+
+    /**
+     * Identify the product from which thumbnail should be taken.
+     *
+     * @param Item $item
+     * @return Product
```
Comment (eduard13) [inline app/code/Magento/Wishlist/Block/Share/Email/Items.php]:
```suggestion
     * @param Item $item
     *
     * @return Product
```

File: app/code/Magento/Wishlist/Block/Share/Email/Items.php [hunk 6]
```diff
@@ -11,17 +11,61 @@
...
+    private $itemResolver;
+
     /**
      * @var string
      */
     protected $_template = 'Magento_Wishlist::email/items.phtml';
 
+    /**
+     * Items constructor.
+     * @param \Magento\Catalog\Block\Product\Context $context
```
Comment (eduard13) [inline app/code/Magento/Wishlist/Block/Share/Email/Items.php]:
```suggestion
     * Items constructor.
     *
     * @param \Magento\Catalog\Block\Product\Context $context
```

**Review by eduard13 — CHANGES_REQUESTED** [review]
Thank you for the adjustment.
Could you please also fix these minor recommendations and fix the failing Static Tests?

**Review by eduard13 — APPROVED** [review]

**Comment by eduard13** [comment]
@magento run Functional Tests CE, Functional Tests B2B, Functional Tests EE

**Comment by magento-engcom-team** [comment]
Hi @eduard13, thank you for the review.
[ENGCOM-7062](https://jira.corp.magento.com/browse/ENGCOM-7062) has been created to process this Pull Request

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
Failed functional tests not related to the changes in this PR

**Pull request merged**